flang/lib/Lower/OpenMP.cpp

```diff
@@ -75,6 +75,38 @@ genOMP(Fortran::lower::AbstractConverter &absConv,
       standaloneConstruct.u);
 }
 
+static void
+genOMP(Fortran::lower::AbstractConverter &absConv,
+       Fortran::lower::pft::Evaluation &eval,
+       const Fortran::parser::OpenMPBlockConstruct &blockConstruct) {
+  const auto &blockDirective =
+      std::get<Fortran::parser::OmpBeginBlockDirective>(blockConstruct.t);
+  const auto &parallelDirective =
+      std::get<Fortran::parser::OmpBlockDirective>(blockDirective.t);
+  if (parallelDirective.v == llvm::omp::OMPD_parallel) {
+    auto &firOpBuilder = absConv.getFirOpBuilder();
+    auto currentLocation = absConv.getCurrentLocation();
+    auto insertPt = firOpBuilder.saveInsertionPoint();
+    llvm::ArrayRef<mlir::Type> argTy;
+    mlir::ValueRange range;
+    // FIXME: This can be done in a better way ?
+    std::array<int32_t, 6> operandSegmentSizes = {0, 0, 0, 0, 0, 0};
```

Comment:
I'm not aware of there being a better way of doing this, I did something very similar in the parallel op parser in MLIR anyway.
That's not to say we shouldn't necessarily add a better way of doing it to MLIR, maybe an overload for getI32VectorAttr that takes an integer and array size instead and uses that. It might be useful if this is going to be a common pattern.

Reply:
`getI32VectorAttr` should be able to take a `std::vector` which can be constructed to a size of 6 and use the default element ctor. Whether that's "better" is in the eye of the beholder, of course.

Reply:
Now that you mention it... Can it not already? getI32VectorAttr takes an ArrayRef and I believe a std::vector is implicitly convertable to an ArrayRef?

Reply:
That is to say I think calling getI32VectorAttr with `std::vector<int64_t>(6)` should work?

Reply:
Thanks @schweitzpgi and @DavidTruby. Refactored expression looks cleaner(to me :)).